### PR TITLE
AM - Lidar Calculations Library, Radius Calculation function

### PR DESCRIPTION
**Pull request by adamm08**
Created and tested the lidar calculations library with the radius calculation function. Test node and launch file included in /test directory.

### CODE REVIEW

File: lib/lidar_calculations.cpp [hunk 1]
```diff
@@ -0,0 +1,67 @@
+#include <iostream>
+#include <vector>
+#include <ros/ros.h>
+#include "lidar_calculations.h"
+#include "lidar_point.h"
+#include <eigen3/Eigen/Dense>
+
+// Default constructor
+lidarCalculations::lidarCalculations() {}
```
Comment (gpearcey) [inline lib/lidar_calculations.cpp]: C++ automatically generates a constructor, so unless you want to do something specific in the constructor, you can remove this.

File: lib/lidar_calculations.cpp [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+double lidarCalculations::calculateRadius(const std::vector<lidarPoint>& points, int min_pts) const {
+    if (points.size() < min_pts) {
+        ROS_WARN("At least 6 points are required to calculate the radius of a cylinder.");
```
Comment (gpearcey) [inline lib/lidar_calculations.cpp]: Replace "6" in the warning string with min_pnts.

File: lib/lidar_calculations.cpp [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    Eigen::Matrix<double, 3, 3> matrixA;
+    double mA_11, mA_12, mA_13, mA_21, mA_22, mA_23, mA_31, mA_32, mA_33 = 0;
+    for (size_t i = 0; i < points.size(); i++) {
+        mA_11 = mA_11 + pow(x_coords[i], 2);
+        mA_12 = mA_12 + (x_coords[i]*y_coords[i]);
+        mA_13 = mA_13 + x_coords[i];
+        // mA_21 = mA_12
+        mA_22 = mA_22 + pow(y_coords[i], 2);
+        mA_23 = mA_23 + y_coords[i];
+        // mA_31 = mA_13
```
Comment (gpearcey) [inline lib/lidar_calculations.cpp]: Can these comments be removed?

**Review by gpearcey — CHANGES_REQUESTED** [review]
Can you do a short page in Gitbook with the link to the calculator site used, any other helpful references and a brief explanation of the math (just enough detail that jordan or I could figure it out after reading for example).

**Comment by adamm08** [comment]
Made all recommended changes.

**Review by gpearcey — APPROVED** [review]
GTM

**Pull request merged**